devito/ops/operator.py

```diff
@@ -26,22 +31,47 @@ def _specialize_iet(self, iet, **kwargs):
         ops_partition = Call(namespace['ops_partition'], Literal('""'))
         ops_exit = Call(namespace['ops_exit'])
 
+        ops_block = OpsBlock('block')
+
         dims = []
+        to_dat = set()
+        for section, trees in find_affine_trees(iet).items():
+            dims.append(len(trees[0].dimensions))
+            symbols = set(FindSymbols('symbolics').visit(trees[0].root))
+            symbols -= set(FindSymbols('defines').visit(trees[0].root))
+            to_dat |= symbols
+
+        to_dat = filter_sorted(to_dat)
+
+        name_to_ops_dat = {}
         pre_time_loop = []
+        for f in to_dat:
+            if f.is_Constant:
+                continue
+
+            pre_time_loop.extend(create_ops_dat(f, name_to_ops_dat, ops_block))
+
         for n, (section, trees) in enumerate(find_affine_trees(iet).items()):
             pre_loop, ops_kernel = opsit(trees, n)
 
             pre_time_loop.extend(pre_loop)
             self._ops_kernels.append(ops_kernel)
-            dims.append(len(trees[n].dimensions))
 
         assert (d == dims[0] for d in dims), \
             "The OPS backend currently assumes that all kernels \
             have the same number of dimensions"
 
+        ops_block_init = Expression(ClusterizedEq(Eq(
```

Comment:
What do you think of creating a function `create_ops_block` into transformer.py, as you did for ops_dat?

Reply:
It's two lines of code, I'm not sure it's worth it